onvm/go.sh

```diff
@@ -48,13 +48,32 @@ fi
 
 shift 3
 
+# Verify that bc is installed
+if [[ -z $(command -v bc) ]]
+then
+    echo "Error: bc is not installed. Install using:"
+    echo "  sudo apt-get install bc"
+    echo "See dependencies for more information"
+    exit 1
+fi
+
 if [ -z "$nf_cores" ]
 then
     usage
 fi
 
+ports_bin=$(echo "obase=2; ibase=16; $ports" | bc)
+count_ports=0
+while [[ $ports_bin -ne 0 ]]
+do
+    if [[ $((ports_bin % 10)) -eq 1 ]]
+    then
+        count_ports=$((count_ports+1))
+    fi
+    ports_bin=$ports_bin/10
+done
```

Comment:
I think you can replace this while loop with one or two lines of bash.  The `bc` command will output a string like `1011`.  If we remove all the `0`s then you can just find the length of the resulting string.  Using `${#ports_bin}` will give you the length of the string.

(this is not particularly important, but I'd rather have one slightly confusing line of code than 8)

Reply:
Changed in most recent commit.